usr/opt/OSCX200/src/util/scripts/xon.man
.\"
.\" *****************************************************************
.\" *                                                               *
.\" *    Copyright (c) Digital Equipment Corporation, 1991, 1994    *
.\" *                                                               *
.\" *   All Rights Reserved.  Unpublished rights  reserved  under   *
.\" *   the copyright laws of the United States.                    *
.\" *                                                               *
.\" *   The software contained on this media  is  proprietary  to   *
.\" *   and  embodies  the  confidential  technology  of  Digital   *
.\" *   Equipment Corporation.  Possession, use,  duplication  or   *
.\" *   dissemination of the software and media is authorized only  *
.\" *   pursuant to a valid written license from Digital Equipment  *
.\" *   Corporation.                                                *
.\" *                                                               *
.\" *   RESTRICTED RIGHTS LEGEND   Use, duplication, or disclosure  *
.\" *   by the U.S. Government is subject to restrictions  as  set  *
.\" *   forth in Subparagraph (c)(1)(ii)  of  DFARS  252.227-7013,  *
.\" *   or  in  FAR 52.227-19, as applicable.                       *
.\" *                                                               *
.\" *****************************************************************
.\"
.\"
.\" HISTORY
.\"
.TH XON 1 "Release 5" "X Version 11"
.SH NAME
xon \- start an X program on a remote machine
.SH SYNOPSIS
.B xon
remote-host [-access] [-debug] [-name window-name] [-nols] [-screen screen-no]
[-user user-name] [command ...]
.SH DESCRIPTION
.I Xon
runs the specified command (default xterm -ls) on the remote machine using
rsh.  Xon passes the DISPLAY, XAUTHORITY and XUSERFILESEARCHPATH environment
variables to the remote command.
.PP
When no command is specified, xon runs 'xterm -ls'.  It additionally
specifies the application name to be 'xterm-\fIremote-host\fP' and the
window title to be '-fIremote-host\fP'.
.PP
Xon can only work when the remote host will allow you to use rsh, by having
an entry in the .rhosts file permitting access.
.SH OPTIONS
.PP
Note that the options follow the remote host name (as they do with rlogin).
.sp
.IP "\fB-access\fP"
Runs xhost locally to add the remote host to the host access list in the X
server.  This won't work unless xhost is given permission to modify the
access list.
.IP "\fB-debug\fP"
Normally, xon disconnects the remote process from stdin, stdout and stderr
to eliminate the daemon processes which usually connect them across the
network.  Specifying the \fB-debug\fP option leaves them connected so that
error messages from the remote execution are sent back to the originating
host.
.IP "\fB-name window-name\fP"
This specifies a different application name and window title for the default
command (xterm).
.IP "\fB-nols\fP"
Normally xon passes the -ls option to the remote xterm; this option
suspends that behaviour.
.IP "\fB-screen screen-no\fP"
This changes the screen number of the DISPLAY variable passed to the remote
command.
.IP "\fB-user user-name\fP"
By default, xon simply uses rsh to connect to the remote machine using the
same user name as on the local machine.  This option cause xon to specify an
alternative user name.  This will not work unless you have authorization to
access the remote account via rsh by placing an appropriate entry in the
remote users .rhosts file.
.SH BUGS
Xon can get easily confused when the remote-host, user-name or various
environment variable values contain white space.
.PP
Xon has no way to send the appropriate X authorization information to the
remote host.


usr/opt/OSCX200/src/util/scripts/xon.sh
#!/bin/sh
# 
# *****************************************************************
# *                                                               *
# *    Copyright (c) Digital Equipment Corporation, 1991, 1994    *
# *                                                               *
# *   All Rights Reserved.  Unpublished rights  reserved  under   *
# *   the copyright laws of the United States.                    *
# *                                                               *
# *   The software contained on this media  is  proprietary  to   *
# *   and  embodies  the  confidential  technology  of  Digital   *
# *   Equipment Corporation.  Possession, use,  duplication  or   *
# *   dissemination of the software and media is authorized only  *
# *   pursuant to a valid written license from Digital Equipment  *
# *   Corporation.                                                *
# *                                                               *
# *   RESTRICTED RIGHTS LEGEND   Use, duplication, or disclosure  *
# *   by the U.S. Government is subject to restrictions  as  set  *
# *   forth in Subparagraph (c)(1)(ii)  of  DFARS  252.227-7013,  *
# *   or  in  FAR 52.227-19, as applicable.                       *
# *                                                               *
# *****************************************************************
#
# HISTORY
# 
# start up xterm (or any other X command) on the specified host
# Usage: xon host [arguments] [command]
case $# in
0)
	echo "Usage: $0 <hostname> [-user user] [-name window-name] [-debug]"
	echo "[-screen screen-number] [command ...]"
	exit 1
	;;
esac
target=$1
shift
label=$target
resource=xterm-$label
rcmd="rsh $target"
case $DISPLAY in
unix:*)
	DISPLAY=`echo $DISPLAY | sed 's/unix//'`
	;;
local:*)
	DISPLAY=`echo $DISPLAY | sed 's/local//'`
	;;
esac
case $DISPLAY in
:*)
	fullname=`hostname`
	hostname=`echo $fullname | sed 's/\..*$//'`
	if [ $hostname = $target -o $fullname = $target ]; then
		DISPLAY=$DISPLAY
		rcmd="sh -c"
	else
		DISPLAY=$fullname$DISPLAY
	fi
	;;
esac
username=
xauth=
case x$XUSERFILESEARCHPATH in
x)
	xpath='HOME=${HOME-`pwd`} '
	;;
*)
	xpath='HOME=${HOME-`pwd`} XUSERFILESEARCHPATH=${XUSERFILESEARCHPATH-"'"$XUSERFILESEARCHPATH"'"} '
	;;
esac
redirect=" < /dev/null > /dev/null 2>&1 &"
command=
ls=-ls
continue=:
while $continue; do
	case $1 in
	-user)
		shift
		username="-l $1"
		label="$target $1"
		rcmd="rsh $target $username"
		shift
		case x$XAUTHORITY in
		x)
			XAUTHORITY="$HOME/.Xauthority"
			;;
		esac
		case x$XUSERFILESEARCHPATH in
		x)
			;;
		*)
			xpath="XUSERFILESEARCHPATH=$XUSERFILESEARCHPATH "
			;;
		esac
		;;
	-access)
		shift
		xhost +$target
		;;
	-name)
		shift
		label="$1"
		resource="$1"
		shift
		;;
	-nols)
		shift
		ls=
		;;
	-debug)
		shift
		redirect=
		;;
	-screen)
		shift
		DISPLAY=`echo $DISPLAY | sed 's/:\\([0-9][0-9]*\\)\\.[0-9]/:\1/'`.$1
		shift
		;;
	*)
		continue=false
		;;
	esac
done
case x$XAUTHORITY in
x)
	;;
x*)
	xauth="XAUTHORITY=$XAUTHORITY "
	;;
esac
vars="$xpath$xauth"DISPLAY="$DISPLAY"
case $# in
0)
	$rcmd 'sh -c '"'$vars"' xterm '$ls' -name "'"$resource"'" -T "'"$label"'" -n "'"$label"'" '"$redirect'"
	;;
*)
	$rcmd 'sh -c '"'$vars"' '"$*$redirect'"
	;;
esac
